### PR TITLE
Remove `ruby-enum` dependency

### LINKED ISSUE #135: Open to removing the ruby-enum dependency?
I'd like to propose removing the `ruby-enum` dependency as it unnecessarily increases the footprint of this gem. `ruby-enum` includes `i18n` as a dependency, which itself depends on `concurrent-ruby`. We appear to only use `ruby-enum` in the config objects and don't use much of the functionality it offers. I'd even be willing to do the work of replacing.

Is this a contribution you'd be interested in seeing?

### PR DESCRIPTION
**Pull request by ojab**
Fixes #135

### CODE REVIEW

**Comment by ojab** [comment]
Sending to review due to `First-time contributors need a maintainer to approve running workflows.`. `bundler exec rake` passes locally.

**Comment by ojab** [comment]
Ugh, CI is failing due to
```
commonmarker.gemspec:24:29: C: Gemspec/RequiredRubyVersion: required_ruby_version (2.4, declared in commonmarker.gemspec) and TargetRubyVersion (2.5, which may be specified in .rubocop.yml) should be equal.
  s.required_ruby_version = ['>= 2.4.10', '< 4.0']
                            ^^^^^^^^^^^^^^^^^^^^^^

**Comment by gjtorikian** [comment]
> Since CI on 2.4 is not running — is it really supported or constraint in gemfile can be bumped to 2.5?

Let's actually bring the constraint to 2.6, and remove 2.5 from CI. 2.5 is EOL and I don't want to support EOL Rubies. Thanks for bringing this up!

**Comment by ojab** [comment]
Done, first commit bumps ruby to 2.6, second actually removing ruby-enum. And again ` First-time contributors need a maintainer to approve running workflows.`.

File: lib/commonmarker/config.rb [hunk 1]
```diff
@@ -1,55 +1,48 @@
...
-      type = Config.const_get(type.capitalize)
+      type = OPTS_TYPES.fetch(type)
       case option
       when Symbol
-        check_option(option, type)
-        type.to_h[option]
+        const_get(type).fetch(option)
       when Array
-        option = [nil] if option.empty?
+        raise TypeError if option.none?
```
[the remark this answers is not in the document]
Comment (gjtorikian) [reply lib/commonmarker/config.rb:35]: Good points. Regarding being "overly strict"—I have a tendency to lean towards paranoia when interactions with C occur, hence the small whitelist of accepted types.

**Comment by gjtorikian** [comment]
Thank you for your contribution here, I really appreciate it.

**Pull request merged**